### PR TITLE
Desktop doc

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,11 @@ Provided Functionality
 
 See [the generated yardoc documentation](http://www.rubydoc.info/github/yast/yast-yast2) at rubydoc.info.
 
+It contain some general overview of yast2 parts:
```
Comment (imobach) [inline README.md]: 'It contains'

File: doc/desktop_file.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Desktop Files
+Desktop files are used to show entries in yast2 control center. They are used
+also by autoYaST to pass information configuration.
```
Comment (imobach) [inline doc/desktop_file.md]: Nitpicking: I'd capitalize 'AutoYaST'

File: doc/desktop_file.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+YaST desktop files are valid desktop files as defined by freedesktop organization.
```
Comment (imobach) [inline doc/desktop_file.md]: Maybe adding some link to the Freedesktop's definition could be great.

File: doc/desktop_file.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+For details about syntax, comments, localization and other stuff see
+[latest desktop entry spec](http://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html).
```
Comment (imobach) [inline doc/desktop_file.md]: Forget my prior comment :)

File: doc/desktop_file.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+## Recognized Standard Keys
+YaST control center require especially these keys:
+
+* *Type* Yast use only `application`. It is mandatory key.
```
Comment (imobach) [inline doc/desktop_file.md]: "a mandatory key".

File: doc/desktop_file.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+* *Name* Module name like `Firewall`. Localized form is supported like
+  `Name[cs_CZ]=Pekelna brana`.
+
+* *GenericName* In YaST used as description of the application purpose,
+  e.g. Configure a firewall.  Localized form is supported like
+  `GenericName[cs_CZ]=Upravit pocet byrokratu u pekelne brany`.
+
+* *Categories* A semicolon-separated list of categories in which the entry will
+  be show in a menu, e.g. `Qt;X-SuSE-YaST;X-SuSE-YaST-Security;` mean that it is
```
Comment (imobach) [inline doc/desktop_file.md]: "be shown"

Comment (imobach) [inline doc/desktop_file.md]: "means"

File: doc/desktop_file.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+* *X-SuSE-YaST-RootOnly* This entry defines whether the application will be
+  visible only for root. Possible values are `true` and `false`.
+
+## AutoYaST Specific Keys
+AutoYaST have own keys that are also enclosed in X-SuSE-YaST namespace and have
+Auto prefix:
+
+* *X-SuSE-YaST-AutoInst* Specifies the module compatibility level with the
+  AutoYaST. If not specified then no autoyast support is availabel, otherwise
```
Comment (imobach) [inline doc/desktop_file.md]: Fix 'availabel' typo.

File: doc/desktop_file.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+* *X-SuSE-YaST-AutoInstPath* Deprecated option. Do nothing.
+
+* *X-SuSE-YaST-AutoInstClient* Name of the client to call. By default it is
+  `{module name}_auto`.
+
+* *X-SuSE-YaST-AutoInstDataType* Data type of configuration section. Possible
+  values are `list` and `map`. Default value is `map`.
+
+* *X-SuSE-YaST-AutoInstResource* Specifies top level xml node under which is located
```
Comment (imobach) [inline doc/desktop_file.md]: I think that 'xml' should be capitalized.

File: doc/desktop_file.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+  module specific configuration in AutoYaST profile.
+
+* *X-SuSE-YaST-AutoInstRequires* Contains comma separated list of modules that
+  are required to run before this module. By default empty.
```
Comment (imobach) [inline doc/desktop_file.md]: "is empty"

File: doc/desktop_file.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+* *X-SuSE-YaST-AutoInstMerge* Contains comma separated list of sections that
+  can be handled by one module. For example users module can handle also groups
+  and user\_defaults. By default empty.
```
Comment (imobach) [inline doc/desktop_file.md]: "is empty"

File: doc/desktop_file.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+* *X-SuSE-YaST-AutoInstMergeTypes* Contains comman separated list of section
+  types. Useful only together with *X-SuSE-YaST-AutoInstMerge*. By default
+  empty.
```
Comment (imobach) [inline doc/desktop_file.md]: "is empty"

File: doc/desktop_file.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+* *X-SuSE-YaST-AutoInstClonable* Specifies if module can be cloned. Possible
+  values are `true` and `false`.
+
+* *X-SuSE-YaST-AutoInstSchema* Specifies base name of schema file, including
+  the rnc extension (Relax NG compact syntax). By default empty.
+
+* *X-SuSE-YaST-AutoInstOptional* Specifies if the element is optional in
```
Comment (imobach) [inline doc/desktop_file.md]: I'm not sure if this 'if' should be 'whether'. Same in line 93.

**Comment by jreidinger** [comment]
@imobach thanks, adapted

**Comment by imobach** [comment]
LGTM

**Comment by imobach** [comment]
LGTM

**Pull request merged**